apis/projectcontour/v1/httpproxy.go

```diff
@@ -815,45 +815,50 @@ type LocalRateLimitPolicy struct {
 
 // GlobalRateLimitPolicy defines global rate limiting parameters.
 type GlobalRateLimitPolicy struct {
+	// DefaultGlobalRateLimitPolicyDisabled configures the HTTPProxy to not use
+	// the default global rate limit policy defined by the Contour configuration.
+	// +optional
+	DefaultGlobalRateLimitPolicyDisabled bool `json:"defaultGlobalRateLimitPolicyDisabled,omitempty"`
+
 	// Descriptors defines the list of descriptors that will
 	// be generated and sent to the rate limit service. Each
 	// descriptor contains 1+ key-value pair entries.
-	// +required
+	// +optional
 	// +kubebuilder:validation:MinItems=1
-	Descriptors []RateLimitDescriptor `json:"descriptors,omitempty"`
+	Descriptors []RateLimitDescriptor `json:"descriptors,omitempty" yaml:"descriptors,omitempty"`
```

Comment:
are the struct tags for yaml required? the crd generator tooling should only need the json ones

Reply:
I also found it annoying that we need to define `yaml` tags here 😞 
The contour's config yaml uses this struct and unfortunately, it cannot parse these structs without adding the yaml tags.

An option here would be to redefine everything in the ContourConfig struct and separate them. wdyt?